platform/darwin/src/MGLOfflineStorage.mm

```diff
@@ -357,6 +360,31 @@ - (void)addPackForRegion:(id <MGLOfflineRegion>)region withContext:(NSData *)con
         [[strongSelf mutableArrayValueForKey:@"packs"] addObject:pack];
         if (completion) {
             completion(pack, error);
+            
+            MGLTilePyramidOfflineRegion *tileRegion = MGL_OBJC_DYNAMIC_CAST(region, MGLTilePyramidOfflineRegion);
+            MGLShapeOfflineRegion *shapeRegion = MGL_OBJC_DYNAMIC_CAST(region, MGLShapeOfflineRegion);
+            
+            NSDictionary *attributes;
+            
+            if ([region isKindOfClass:[MGLTilePyramidOfflineRegion class]]) {
```

Comment:
Since you already have `tileRegion` and `shapeRegion` you could just check `if (tileRegion) ...` here.